teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -630,16 +593,62 @@ spec:
       value: 0.3
 ```
 
-#### HTTP Endpoint
+As we are working on migrating from the Tekton Hub to the Artifact Hub ([tektoncd/hub#667][667]), we propose adding an optional `"type"` parameter to the Hub Resolver indicating which Hub API should be used to fetch the catalog. The value can be set to `"artifact"` or `"tekton"`. The default value is `"artifact"`.
 
-In [TEP-0110][tep-0110], we added an HTTP endpoint that serves Tekton resources that users can apply directly to their
-clusters. These endpoints should continue working as is:
+Note that the Catalog versioning in the Artifact Hub follows the semantic versioning (i.e. <major-version>.<minor-version>.0). For backward compatibility in the Hub Resolver, both full (i.e. <major-version>.<minor-version>.0) and simplified semantic versioning (i.e. <major-version>.<minor-version>) will be accepted by the `"version"` parameter. The Hub Resolver will map the version to the format expected by the target Hub type.
 
+Example:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: openshift-install
+spec:
+  taskRef:
+    resolver: hub
+    resource:
+    - name: type
+      value: artifact
+    - name: kind
+      value: task
+    - name: name
+      value: openshift-client
+    - name: catalog
+      value: openshift
+    - name: version
+      value: 0.3
+```
+
+
+When setting to `"artifact"`, the following Artifact Hub API will be invoked to fetch **task** resources:
+```shell
+curl -X 'GET' \
+  '{ARTIFACT_HUB_API}/api/v1/packages/tekton-<resource-kind>/<catalog-name>/<resource-name>/<resource-version>' \
+  -H 'accept: application/json'
+```
+
+The `ARTIFACT_HUB_API` can be configured by setting the `ARTIFACT_HUB_API` environment variable in [resolver-deployment.yaml](https://github.com/tektoncd/pipeline/blob/main/config/resolvers/resolvers-deployment.yaml). The default value is `https://artifacthub.io/`.
+```yaml
+env
+- name: ARTIFACT_HUB_API
+  value: "https://artifacthub.io/"
+```
+
+When setting to `"tekton"`, the following Artifact Hub API will be invoked to fetch resources:
 ```shell
-kubectl apply -f https://api.hub.tekton.dev/v1/resource/tekton/task/buildpacks/0.5/raw
+curl -X 'GET' \
+  '{TEKTON_HUB_API}/v1/resource/<catalog-name>/<resource-kind>/<resource-name>/<resource-version>/yaml'
+```
+
+The `TEKTON_HUB_API` can be configured by setting the `TEKTON_HUB_API` environment variable in [resolver-deployment.yaml](https://github.com/tektoncd/pipeline/blob/main/config/resolvers/resolvers-deployment.yaml). For backward compatability of the Hub Resolver, we propose to remove the default value (`https://api.hub.tekton.dev`) of `TEKTON_HUB_API` since the default `type` is changed to `artifact` and the Tekton Hub is planned to be deprecated soon. The users must configure their own private instance of the Tekton Hub to use the Hub Resolver with `"type"` setting to `"tekton"`.
```

Comment:
Hi @vdemeester. I did a minor update here to remove the default value of `TEKTON_HUB_API` for backward compatibility after deprecating https://api.hub.tekton.dev. Please take a look 🙇